From 326ac4a3df869abe851ecfceedf9d99e4b89a594 Mon Sep 17 00:00:00 2001
From: Lily Chen <chlily@chromium.org>
Date: Wed, 07 Jun 2023 21:42:15 +0000
Subject: [PATCH] [M114][DownloadBubble] Clear default button when buttons are cleared

This clears the default button on the bubble dialog delegate when the
buttons are cleared (set to DIALOG_BUTTON_NONE). This avoids tripping
a CHECK where we crash when the default button is not among the buttons
present on the dialog.

(cherry picked from commit a339a05569b03d0bb5be2f5a8fc239734adbabe6)

Bug: 1449354
Change-Id: Ia5064b80e0c8c56c08b69ee8df6b085436ffa267
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4574027
Commit-Queue: Lily Chen <chlily@chromium.org>
Reviewed-by: Daniel Rubery <drubery@chromium.org>
Cr-Original-Commit-Position: refs/heads/main@{#1150741}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4599229
Auto-Submit: Lily Chen <chlily@chromium.org>
Commit-Queue: Daniel Rubery <drubery@chromium.org>
Cr-Commit-Position: refs/branch-heads/5735@{#1189}
Cr-Branched-From: 2f562e4ddbaf79a3f3cb338b4d1bd4398d49eb67-refs/heads/main@{#1135570}
---

--- a/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc
+++ b/chrome/browser/ui/views/download/bubble/download_toolbar_button_view.cc
@@ -419,6 +419,7 @@
   primary_view_->SetVisible(true);
   security_view_->SetVisible(false);
   bubble_delegate_->SetButtons(ui::DIALOG_BUTTON_NONE);
+  bubble_delegate_->SetDefaultButton(ui::DIALOG_BUTTON_NONE);
   bubble_delegate_->set_margins(GetPrimaryViewMargin());
   ResizeDialog();
 }
@@ -468,6 +469,7 @@
   bubble_delegate->SetShowTitle(false);
   bubble_delegate->SetShowCloseButton(false);
   bubble_delegate->SetButtons(ui::DIALOG_BUTTON_NONE);
+  bubble_delegate->SetDefaultButton(ui::DIALOG_BUTTON_NONE);
   bubble_delegate->RegisterDeleteDelegateCallback(
       base::BindOnce(&DownloadToolbarButtonView::OnBubbleDelegateDeleted,
                      weak_factory_.GetWeakPtr()));
